From 55d0e1e5a1e647321c13b3d346efbc7556def75c Mon Sep 17 00:00:00 2001
From: Benjamin Leggett <benjamin@edera.io>
Date: Fri, 3 Oct 2025 23:06:24 -0400
Subject: [PATCH] Use built-in PAT even on x86

---
 kernel-open/nvidia/nv-pat.c | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/kernel-open/nvidia/nv-pat.c b/kernel-open/nvidia/nv-pat.c
index 870dedbd..5a43bf21 100644
--- a/kernel-open/nvidia/nv-pat.c
+++ b/kernel-open/nvidia/nv-pat.c
@@ -40,12 +40,9 @@ int nv_pat_mode = NV_PAT_MODE_DISABLED;
  * WC entry is as expected before using PAT.
  */
 
-#if defined(CONFIG_X86_PAT)
-#define NV_ENABLE_BUILTIN_PAT_SUPPORT 0
-#else
+// HACK(edera) - force use of builtin PAT even on x86 to work around
+// buggy PAT detection under Xen
 #define NV_ENABLE_BUILTIN_PAT_SUPPORT 1
-#endif
-
 
 #define NV_READ_PAT_ENTRIES(pat1, pat2)   rdmsr(0x277, (pat1), (pat2))
 #define NV_WRITE_PAT_ENTRIES(pat1, pat2)  wrmsr(0x277, (pat1), (pat2))
@@ -63,14 +60,14 @@ static inline void nv_disable_caches(unsigned long *cr4)
     wbinvd();
     *cr4 = __read_cr4();
     if (*cr4 & 0x80) __write_cr4(*cr4 & ~0x80);
-    __flush_tlb();
+    __flush_tlb_local();
 }
 
 static inline void nv_enable_caches(unsigned long cr4)
 {
     unsigned long cr0 = read_cr0();
     wbinvd();
-    __flush_tlb();
+    __flush_tlb_local();
     write_cr0((cr0 & 0x9fffffff));
     if (cr4 & 0x80) __write_cr4(cr4);
 }
@@ -317,8 +314,12 @@ static int nv_determine_pat_mode(void)
     else if (PAT_WC_index != 0xf)
     {
         nv_printf(NV_DBG_ERRORS,
-            "NVRM: PAT configuration unsupported.\n");
-        return NV_PAT_MODE_DISABLED;
+            "NVRM: Hardware PAT configuration unsupported, using builtin.\n");
+        #if NV_ENABLE_BUILTIN_PAT_SUPPORT
+                return NV_PAT_MODE_BUILTIN;
+        #else
+                return NV_PAT_MODE_DISABLED;
+        #endif /* NV_ENABLE_BUILTIN_PAT_SUPPORT */
     }
     else
     {
-- 
2.51.0

